react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js

```diff
@@ -15,24 +18,36 @@ class App extends Component {
   };
 
   componentDidMount() {
-    store.subscribe(() => {
-      const { books, bookSelected } = store.getState();
-      this.setState({ books, bookSelected });
-    });
-    // TODO to implement the dispatch
+    this.getBooks();
   }
 
-  // TODO to implement the dispatch
-  onSearch = value => {};
+  onSearch = value => {
+    this.props.searchBook(value);
+  };
+
+  getBooks = () => setTimeout(() => this.props.getBooks(), Math.random() * 100);
 
-  // TODO to implement the dispatch
-  addToCart = item => {};
+  addToCart = item => {
+    this.props.addToCart(item);
+  };
 
-  // TODO to implement the dispatch
-  addItem = itemId => {};
+  addItem = itemId => {
+    this.props.addItem(itemId);
+  };
 
-  // TODO to implement the dispatch
-  removeItem = itemId => {};
+  subtractItem = itemId => {
+    const { bookSelected } = this.props;
+    const bookS = bookSelected.find(book => book.id === itemId);
+    if (bookS.quantity > 1) {
```

Comment:
i dont know if this is correct, be careful because if  `books` find is null this will gonna break.
Also `.find` gives you only the first item that satisfy the condition, so `bookS` is not a correct name for this.